harness/determined/common/experimental/model.py

```diff
@@ -301,29 +305,32 @@ def set_labels(self, labels: List[str]) -> None:
         if not isinstance(labels, Iterable):
             raise ValueError(f"set_labels() requires a list of strings as input but got: {labels}")
 
-        self.labels = list(labels)
+        labels = list(labels)
 
-        req = bindings.v1PatchModel(labels=self.labels)
+        req = bindings.v1PatchModel(labels=labels)
         bindings.patch_PatchModel(self._session, body=req, modelName=self.name)
 
+        self.labels = labels
```

Comment:
same comment as the metadata, should we update the state with the PATCH response?